module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -166,7 +164,10 @@ func (ir *IncorporatedResultSeals) RegisterEjectionCallbacks(callbacks ...mempoo
 func (ir *IncorporatedResultSeals) PruneUpToHeight(height uint64) error {
 	return ir.Backend.Run(func(entities map[flow.Identifier]flow.Entity) error {
 		if height < ir.lowestHeight {
-			return fmt.Errorf("new pruning height %v cannot be smaller than previous pruned height:%v", height, ir.lowestHeight)
+			// when it was called concurrently with different heights, the caller can't ensure the strict
+			// order of heights passed, in that case, we will just ignore the height instead
+			// of returning an error.
+			return nil
```

Comment:
It would be great to return a sentinel error here. The Higher-level logic can then decide, based on how the `IncorporatedResultSeals` is used, whether out-of-order pruning should be an error or is expect:
```suggestion
			//pruned height cannot decrease, as we cannot recover already pruned elements
			return errors.NewDecreasingPruningHeightError("new pruning height %v cannot be smaller than previous pruned height:%v", height, ir.lowestHeight)
```



As our current logic in `sealing.Core` is fully concurrent, we would expect the sentinel error to occur.